qt/scientific_interfaces/Indirect/IndirectSettingsPresenter.h

```diff
@@ -12,41 +12,32 @@
 
 #include <memory>
 
-#include <QObject>
-#include <QVariant>
-
 namespace MantidQt {
 namespace CustomInterfaces {
+class IIndirectSettings;
 
-class MANTIDQT_INDIRECT_DLL IndirectSettingsPresenter : public QObject {
-  Q_OBJECT
+class MANTIDQT_INDIRECT_DLL IndirectSettingsPresenter {
 
 public:
-  explicit IndirectSettingsPresenter(QWidget *parent = nullptr);
-  explicit IndirectSettingsPresenter(IndirectSettingsModel *model, IIndirectSettingsView *view);
+  explicit IndirectSettingsPresenter(std::unique_ptr<IndirectSettingsModel> model, IIndirectSettingsView *view);
 
-  IIndirectSettingsView *getView();
+  QWidget *getView();
+  void subscribeParent(IIndirectSettings *parent);
 
   void loadSettings();
 
-signals:
-  void closeSettings();
-  void applySettings();
-
-private slots:
-  void applyAndCloseSettings();
-  void applyChanges();
-  void closeDialog();
+  void notifyOkClicked();
+  void notifyApplyClicked();
+  void notifyCancelClicked();
 
 private:
-  void setUpPresenter();
-  void setDefaultRestrictData() const;
   void saveSettings();
 
   void setApplyingChanges(bool applyingChanges);
 
   std::unique_ptr<IndirectSettingsModel> m_model;
-  std::unique_ptr<IIndirectSettingsView> m_view;
+  IIndirectSettingsView *m_view;
```

Comment:
As I am quite new to C++, I am going to be asking some questions that might not make much sense. These questions are for me to learn rather than to suggest a change to the code. Okay so, in this particular case, you changed a unique pointer to the view to a normal pointer, but kept a unique pointer to the model. Does this have anything to do with the fact that the view inherits form QWidget?

Reply:
Yes it does indeed. All of the QWidgets are owned by the QApplication, and so the QApplication handles their lifetime and makes sure they get deleted. This means we do not want to use an owning-pointer such as a unique pointer in our presenter. Owning a QWidget in two places may cause issues if the widget gets cleaned up from one location, but not the other